src/plugin/systems/multiple_rapier_contexts.rs

```diff
@@ -0,0 +1,220 @@
+//! systems to support multiple worlds, and changes between them.
+
+use crate::dynamics::{
+    RapierImpulseJointHandle, RapierMultibodyJointHandle, RapierRigidBodyHandle,
+};
+use crate::geometry::RapierColliderHandle;
+use crate::plugin::{RapierContext, RapierContextEntityLink};
+use bevy::prelude::*;
+
+/// If an entity is turned into the child of something with a physics context link,
+/// the child should become a part of that physics context
+///
+/// If this fails to happen, weirdness will ensue.
+pub fn on_add_entity_with_parent(
+    q_add_entity_without_parent: Query<
+        (Entity, &Parent),
+        (
+            With<RapierContextEntityLink>,
+            Or<(Changed<RapierContextEntityLink>, Changed<Parent>)>,
+        ),
+    >,
+    q_parent: Query<&Parent>,
+    q_physics_world: Query<&RapierContextEntityLink>,
+    mut commands: Commands,
+) {
+    for (ent, parent) in &q_add_entity_without_parent {
+        let mut parent = Some(parent.get());
+        while let Some(parent_entity) = parent {
+            if let Ok(pw) = q_physics_world.get(parent_entity) {
```

Comment:
We should ensure the world is actually changing before inserting it - I've noticed this causes a chain of unending world changes.

```rs
if let Ok(pw) = q_physics_world.get(parent_entity) {
	if q_physics_world.get(ent).map(|x| x != pw).unwrap_or(false) {
		info!("Removing old physx for {ent:?}!");
		remove_old_physics(ent, &mut commands);
		commands.entity(ent).insert(*pw);
	}
	break;
}
```

Reply:
Woops! That unwrap_or should be `unwrap_or(true)` - my bad!